app/screens/ChatbotScreen.tsx

```diff
@@ -0,0 +1,361 @@
+          // jump to last index
+          setPageIndex(combinedPages.length - 1)
+          setLoading(false)
+        }
+        return combinedPages
+      })
+  }
+
+  const requestEditAnswer = (newValue: string, timestamp: number, chatbotId?: string) => {
+    setRequestingNextStep(true)
+    setLoading(true)
+    requestSkip(timestamp, sessionId, chatbotId)
+      .then(() => {
+        // request update to restore all pages from KYC spider
+        return restoreSession(sessionId, chatbotId, [], false)
+          
+      })
+      .then((newPages) => {
+        // given answer is not attached to any page anymore, because of recreation of all pages
+        // therefore get recreated pages' last page answer object and fill that with correct values via requestNextStep
+        let recreatedAnswer = newPages[newPages.length-1].answer
+        if (recreatedAnswer !== undefined) {
+          requestNextStep(chatbotCreateAnswer(newValue, recreatedAnswer), recreatedAnswer, chatbotId, newPages)
+        }
+      })
+  }
+
+  const requestNextStep = (apiAnswer: ChatbotAPIAnswer, answer?: ChatbotAnswer, chatbotId?: string, inputPages?: ChatbotPage[]): Promise<ChatbotPage[]> => {
+    setRequestingNextStep(true)
+    if (inputPages === undefined) {
+      inputPages = pages
+    }
+    return nextStep(sessionId, chatbotId, apiAnswer)
+      .then((question) => {
+        setRequestingNextStep(false)
+        if (answer !== undefined) {
+          chatbotFillAnswerWithData(question, answer)
+        }
+        let [newPages, isFinished, help] = chatbotFeedQuestion(question, settings?.language)
+        let combinedPages: ChatbotPage[] = []
+        if (help !== undefined) {
+          NotificationService.error(help)
+        } else {
+          if (newPages.length > 0) {
+            let combinedPages = inputPages?.concat(newPages) ?? newPages
+            setPages(combinedPages)
+            onNext(combinedPages, isFinished, answer)
+          } else {
+            NotificationService.error(t("kyc.bot.error.validation"))
+          }
+        }
+        setLoading(false)
+        return combinedPages
+      })
+  }
+
+  const updateProgress = (pages: ChatbotPage[], pageIndex: number, isBack: boolean = false) => {
+    if (pageIndex <= 0) {
+      setProgress(0)
+    }
+    // Krysh: there are always around 20 questions. might be more in some cases.
+    // current page / max pages
+    let newProgress = Math.max(pageIndex, 1) / Math.max(pages.length, 20)
+    // on back allow lower progress to previous one
+    if (isBack) {
+      setProgress(newProgress)
+    } 
+    // otherwise only higher progress are allowed, to avoid any confusion for users
+    else if (newProgress > progress) {
+      setProgress(newProgress)
+    }
+  }
+
+  return (
+    <View style={styles.container}>
+      {/* Loading */}
+      {isLoading && <Loading size="large" />}
+      {/* SMS screen */}
+      {!isLoading && !isSMSCompleted && (
```

Comment:
how about?

```
{isLoading 
   ? 
        (<Loading size="large" />)
   :   
        {/* SMS screen */}
        {!isSMSCompleted && (
}
```

Reply:
fine for me, in the beginning it was easier for to read it.

Reply:
nvm it will get too cluttered
```
{isLoading 
   ? 
        (<Loading size="large" />)
   :   
        {!isSMSCompleted
                 ?
                        ({/* SMS screen */})
                 :
                        ({/* Chatbot page screen */})
}
```

I will keep it for now and revisit it with next iteration of chatbot